### PR TITLE
refactor(examples) Update fl-dp-sa example

### PR DESCRIPTION
**Pull request by mohammadnaseri**

### CODE REVIEW

File: examples/fl-dp-sa/fl_dp_sa/task.py [hunk 1]
```diff
@@ -13,10 +13,10 @@
 
 DEVICE = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
```
Comment (jafermarq) [inline examples/fl-dp-sa/fl_dp_sa/task.py:14]: this shouldn't de defined as a global variable. It's better for simulations if clients "auto-discover" the device at run time, like this: https://github.com/adap/flower/blob/486cac3c5e68f717ad023e87e01bbb87c115c37d/examples/quickstart-pytorch/pytorchexample/client_app.py#L18

File: examples/fl-dp-sa/pyproject.toml [hunk 2]
```diff
@@ -1,21 +1,39 @@
...
-torchvision = "0.17.1"
+[tool.hatch.build.targets.wheel]
+packages = ["."]
+
+[tool.flwr.app]
+publisher = "flwrlabs"
+
+[tool.flwr.app.components]
+serverapp = "fastai_example.server_app:app"
+clientapp = "fastai_example.client_app:app"
```
Comment (jafermarq) [inline examples/fl-dp-sa/pyproject.toml:25]:
```suggestion
serverapp = "fl_dp_sa.server_app:app"
clientapp = "fl_dp_sa.client_app:app"
```

File: examples/fl-dp-sa/fl_dp_sa/client_app.py [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
         return loss, len(self.testloader.dataset), {"accuracy": accuracy}
 
 
-def client_fn(cid: str):
-    """Create and return an instance of Flower `Client`."""
-    trainloader, testloader = load_data(partition_id=int(cid))
+def client_fn(context: Context):
+    partition_id = context.node_config["partition-id"]
+    trainloader, testloader = load_data(
+        partition_id=partition_id, num_partitions=context.node_config["num-partitions"]
```
Comment (jafermarq) [inline examples/fl-dp-sa/fl_dp_sa/client_app.py:32]: your `load_data()` function isn't ready to receive the number of partitions.

**Review by jafermarq — COMMENTED** [review]
@mohammadnaseri , my first part of the review is in. Some changes are needed before the code can run. See comments about it below.

Also, let's remove the `requierements.txt` 

Update: let's include `DP` and `SecAgg` to the `[tags]` block at the top of the readme? and remove `basic`.

File: examples/fl-dp-sa/fl_dp_sa/task.py [hunk 4]
```diff
@@ -13,10 +13,10 @@
...
 
+fds = None  # Cache FederatedDataset
```
Comment (jafermarq) [inline examples/fl-dp-sa/fl_dp_sa/task.py:16]: (forgot to add this comment in my review before finalising it)

This `fds` isn't used. You'll need to make use of it similarly to how other `quickstart` examples or migrated examples in general use it. See this: https://github.com/adap/flower/blob/486cac3c5e68f717ad023e87e01bbb87c115c37d/examples/quickstart-pytorch/pytorchexample/task.py#L51

File: examples/fl-dp-sa/fl_dp_sa/server_app.py [hunk 5]
```diff
@@ -31,30 +32,36 @@ def weighted_average(metrics: List[Tuple[int, Metrics]]) -> Metrics:
...
 
+    # Initialize global model
+    model_weights = get_weights(Net())
+    parameters = ndarrays_to_parameters(model_weights)
+
+    strategy = FedAvg(
+        fraction_fit=0.2,
+        fraction_evaluate=0.0,
+        min_fit_clients=20,
+        min_available_clients=20,
```
Comment (jafermarq) [inline examples/fl-dp-sa/fl_dp_sa/server_app.py:49]: do we need this?

File: examples/fl-dp-sa/fl_dp_sa/server_app.py [hunk 6]
```diff
@@ -31,30 +32,36 @@ def weighted_average(metrics: List[Tuple[int, Metrics]]) -> Metrics:
...
+        fit_metrics_aggregation_fn=weighted_average,
+        initial_parameters=parameters,
+    )
 
-app = ServerApp()
+    noise_multiplier = context.run_config["noise-multiplier"]
+    clipping_norm = context.run_config["clipping-norm"]
+    num_sampled_clients = context.run_config["num-sampled-clients"]
```
Comment (jafermarq) [inline examples/fl-dp-sa/fl_dp_sa/server_app.py:56]: Is it good if this parameter isn't linked to `fraction-fit` and in general the total number of nodes? Should this be mentioned somewhere in the example?

File: examples/fl-dp-sa/fl_dp_sa/server_app.py (hunk 5 above)
Comment (mohammadnaseri) [reply examples/fl-dp-sa/fl_dp_sa/server_app.py]: I deleted it.

File: examples/fl-dp-sa/fl_dp_sa/server_app.py (hunk 6 above)
Comment (mohammadnaseri) [reply examples/fl-dp-sa/fl_dp_sa/server_app.py:56]: I added a note for clarifying.

**Review by jafermarq — APPROVED** [review]

**Pull request merged**